### PR TITLE
test: login with federated token for ACR

### PR DESCRIPTION
**Pull request by nojnhuh**
#### What type of PR is this?
/kind cleanup
/kind failing-test

#### What this PR does / why we need it:

This PR modifies an `az login` command invoked from the e2e tests to use a federated token when available instead of always trying to use a service principal password.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/utils/container_registry_utils.go [hunk 1]
```diff
@@ -88,11 +90,30 @@ func AZACRLogin() (err error) {
...
+		if err != nil {
+			return err
+		}
+		authFlags = []string{
+			"--username", authConfig.AADClientID,
+			"--federated-token", string(token),
+		}
+	}
+	args := []string{
+		"login", "--service-principal",
```
Comment (nilo19) [inline tests/e2e/utils/container_registry_utils.go:112]: Do we still need "--service-principal" here

**Comment by nilo19** [comment]
@feiskyer we can ignore the errors as they will be fixed by https://github.com/kubernetes-sigs/cloud-provider-azure/pull/6673

File: tests/e2e/utils/container_registry_utils.go (hunk 1 above)
Comment (nojnhuh) [reply tests/e2e/utils/container_registry_utils.go:112]: I think so, yes. The federated token is still tied to a Service Principal, but that Service Principal doesn't have any passwords. That's my understanding of how things are set up now.

**Comment by nojnhuh** [comment]
@nilo19 It looks like the `eks-test` job keeping this PR and #6673 from merging. Should we mark that job optional in test-infra or `/override` it in these PRs or make both changes in the same PR?

**Pull request merged**